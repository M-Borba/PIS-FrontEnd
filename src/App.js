import React, { useState } from "react";
import PersonView from "./containers/PersonView";
import ProjectView from "./containers/ProjectView";
import { NOT_LOGGED } from "./config/globalVariables";
import {
  BrowserRouter as Router,
  Switch as SwitchRouter,
  Route,
  Redirect,
} from "react-router-dom";
import LoginView from "./containers/Login";
import Header from "./components/Header";
import ListarAdministradores from "./containers/ListarAdministradores";
import ListarPersonas from "./containers/ListarPersonas";
import ListarProyectos from "./containers/ListarProyectos";

export default function App() {
  var uid = localStorage.getItem("uid");
  if (uid == null) {
    uid = NOT_LOGGED;
  }
  const [isProjectView, setIsProjectView] = useState(false);
  const onSwitch = () => {
    setIsProjectView(!isProjectView);
  };

  return (
    <Router>
      <div>
        <Route
          render={({ location }) =>
            !["/login", "/Login"].includes(location.pathname) && <Header />
          }
        />
        <SwitchRouter>
          <Route path="/login" component={LoginView} />
          {uid == NOT_LOGGED && <Redirect to="/login" />}
          <Route path="/personas" component={ListarPersonas} />
          <Route path="/proyectos" component={ListarProyectos} />
          <Route path="/administradores" component={ListarAdministradores} />
          <Route path={["/", "/inicio"]}>
            <>
              <PersonView onSwitch={onSwitch} isProjectView={isProjectView} />
              <ProjectView onSwitch={onSwitch} isProjectView={isProjectView} />
            </>
          </Route>
        </SwitchRouter>
      </div>
    </Router>
  );
}
