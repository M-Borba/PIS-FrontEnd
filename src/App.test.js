import React from "react";
import App from "./App";
import { render, fireEvent } from "@testing-library/react";

test("Logout button renders with correct text", () => {
  localStorage.setItem("uid", "Test");
  const { getByTestId } = render(<App />);
  const logoutButton = getByTestId("logout");

  expect(logoutButton.textContent).toBe("Cerrar Sesion");
  localStorage.removeItem("uid");
});

test("Logout button redirect to /Logout and remove token from local storage", () => {
  localStorage.setItem("uid", "Test");
  localStorage.setItem("client", "Test");
  localStorage.setItem("token", "Test");
  const { getByTestId } = render(<App />);
  const logoutButton = getByTestId("logout");

  fireEvent.click(logoutButton);

  expect(location.pathname).toBe("/Logout");
  expect(localStorage.getItem("uid")).toBe(null);
  expect(localStorage.getItem("client")).toBe(null);
  expect(localStorage.getItem("token")).toBe(null);
});

test("The login is rendered after clicking the logout button", () => {
  localStorage.setItem("uid", "Test");
  const component = render(<App />);
  const logoutButton = component.getByTestId("logout");

  fireEvent.click(logoutButton);

  const login = component.getByTestId("login");
  expect(login).toBeInTheDocument();
});
