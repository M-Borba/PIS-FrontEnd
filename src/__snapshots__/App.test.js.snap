// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          Inicio de sesión
        </div>
      </header>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-7"
      >
        <img
          alt="effectus wallpaper"
          class="logo"
          src="effectus_wallpaper.png"
          width="100%"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-5"
      >
        <div
          class="makeStyles-paper-5"
          data-testid="login"
        >
          <h1
            class="MuiTypography-root MuiTypography-h5"
          >
            Iniciar sesión
          </h1>
          <form
            class="makeStyles-form-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
                data-shrink="true"
                for="email"
                id="email-label"
              >
                Email Address
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="email"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="email"
                  name="email"
                  required=""
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                  >
                    <span>
                      Email Address
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-11"
                  >
                    <span>
                      Password
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-7 MuiButton-containedPrimary MuiButton-fullWidth"
              role="submit"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Sign in
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <h2
              class="MuiTypography-root makeStyles-errorMsg-8 MuiTypography-body1"
            />
            <div
              class="MuiBox-root MuiBox-root-13"
            />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
